utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/source/delayed/DelayedUniversalKafkaSourceFactory.scala

```diff
@@ -73,4 +80,22 @@ class DelayedUniversalKafkaSourceFactory(
       prepareSourceFinalErrors(context, dependencies, step.parameters, errors = Nil)
   }
 
+  protected def timestampFieldParamStep(implicit nodeId: NodeId): NodeTransformationDefinition = {
+    case step @ TransformationStep(
+          (`topicParamName`, DefinedEagerParameter(topic: String, _)) ::
+          (SchemaVersionParamName, DefinedEagerParameter(version: String, _)) :: Nil,
+          _
+        ) =>
+      val preparedTopic = prepareTopic(topic)
+      val versionOption = parseVersionOption(version)
+      val typingResultValidationResult = determineSchemaAndType(
```

Comment:
We have such a concept as a `TransformationStep.state`. Let's put this information into it during and pass it between steps. Thanks to that we avoid unnecessary requests to the schema registry.

Reply:
wasnt sure how to properly do validations before last step so I put in state whole valueSchemaValidationResult

Reply:
Does this state-related approach reduce unnecessary requests to the schema registry? :-)